gdk4/src/keys.rs

```diff
@@ -4662,3 +4662,46 @@ impl IntoGlib for Key {
         self.0
     }
 }
+
+impl ValueType for Key {
+    type Type = u32;
+}
+
+unsafe impl<'a> FromValue<'a> for Key {
+    type Checker = glib::value::NopChecker;
```

Comment:
You need to check if it has the correct `GType` at least. Right now you could get a `gdk::Key` out of a value containing e.g. a widget :P